### ISSUE TITLE
Systemjs interop with UMD bundles, named exports and function default export

### ISSUE THREAD

**Issue by Sauloxd** [systemjs/systemjs#2184]
### Demonstration

Code Sandbox: https://codesandbox.io/s/rough-wind-jzpop?file=/index.html

### Expected Behavior
When one `System.import('library-with-umd-bundle')` it should correctly expose named exports besides the default export.
In this case in specific, we are dealing with `styled-component` library. It exposes a deafult function and lots of named exports:
``` js
import styled, { createGlobalStyle, withTheme } from 'styled-components';
const MyDiv = styled.div`color: blue;`;
const MyGlobalStyle = createGlobalStyle`body { color: red; } `;
```
It's expected that a systemjs bundle with styled-component as dependecy should correctly inject it, supporting the above syntax. 

### Actual Behavior
In the sandbox, I'm using:
- import-map to resolve the UMD bundle for `styled-components`
- import-map to resolve a local `systemjs bundle with styled-component as external` which logs the result of the above syntax

The log shows that the `createGlobalStyle` is undefined, unless you access it via `styled.createGlobalStyle`.

### What I've discussed so far with @joeldenning 
I'm not sure if this is actually a bug or a intended not supported feature 🤔
But Joel pin pointed this line on the [named-exports.js:45](https://github.com/systemjs/systemjs/blob/c977894618c83928c0b843bf998f97bf2e0ffb1c/src/extras/named-exports.js#L45) extra, which only covers `object` default exports.
If we look how `styled-components` expose those named exports ([bundle](https://cdn.jsdelivr.net/npm/styled-components@5.1.0/dist/styled-components.js))
``` js
 var styled = function styled(tag) {
    return constructWithOptions(createStyledComponent, tag);
  }; // Shorthands for all valid HTML Elements
 ...
  for (var key in secondary) {
    styled[key] = secondary[key];
  }

  return styled;
```
It is adding the named exports in styled function itself.

So, should we just enhance `named-exports` extra to expose named exports from default function also? (afaik I don't think there are other enumerables keys in a function, besides the one you add yourself... but can libraries misuse this feature, exposing not intended named exports?)

----- 
Thanks in advance! 😁

**Comment by joeldenning** [comment]
Thanks for the summary @Sauloxd - yes the question here is whether we should extend the named-exports check to allow for functions that have enumerable properties on them.

https://github.com/systemjs/systemjs/blob/c977894618c83928c0b843bf998f97bf2e0ffb1c/src/extras/named-exports.js#L45

Normal functions have no enumerable properties, from what I can tell. So I think that any enumerable properties on functions must be placed there manually, and therefore would be good candidates for making them named exports?

@guybedford could you weigh in on this? Is there a specific reason to limit the named-exports extra to only objects?

**Comment by guybedford** [comment]
Thanks for sharing the feedback here @Sauloxd I think it would be fine to remove the object restriction at this point actually.

We should probably also fixup the iteration to include a hasOwnProperty check too.

It seems like Node.js is making some progress on named exports for CJS, so we should even be able to get this into core by default in the next month or so as well if that moves forward.

**Comment by joeldenning** [comment]
@Sauloxd do you have interest in submitting a PR for this? I think the change could be something similar to this?

```js
if (hasDefaultExport)
  for (var exportName in defaultExport) {
    // default is not a named export
      if (defaultExport.hasOwnProperty(exportName) && exportName !== 'default') {
        _export(exportName, defaultExport[exportName]);
      }
  }
```

^ This plus a test or two would be great. You should run the build before the tests, but then git checkout dist to undo any changes to dist before creating the PR.

**Comment by Sauloxd** [comment]
Would be awesome to contribute here! I will work on it :) thanks guys!

**Comment by joeldenning** [comment]
Resolved in #2186. Will publish in the next few days - am interested to see where #2187 goes first.
